packages/cli/src/commands/deploy/index.ts

```diff
@@ -81,25 +81,13 @@ export default async (client: Client): Promise<number> => {
   let argv = null;
 
   const argOptions: {
-    [k: string]:
-      | BooleanConstructor
-      | StringConstructor
-      | string
-      | [StringConstructor];
+    [k: string]: PrimitiveConstructor | [PrimitiveConstructor] | string;
   } = {};
   for (const option of deployCommand.options) {
-    argOptions[`--${option.name}`] =
-      option.type === 'boolean' ? Boolean : String;
+    argOptions[`--${option.name}`] = option.type;
     if (option.shorthand) {
       argOptions[`-${option.shorthand}`] = `--${option.name}`;
     }
-    if (
-      option.name === 'env' ||
-      option.name === 'build-env' ||
-      option.name === 'meta'
-    ) {
-      argOptions[`--${option.name}`] = [String];
-    }
```

Comment:
**comment (non-blocking):** In future PRs, I would add a PR comment to code like this making it clear why this condition could be removed.

It took me a few minutes to figure out why this change was here and useful.‏

Reply:
I'll do that here by replying to your comment:

In this case, I can remove the extra condition since the `CommandOption` can now take constructors and `[Constructor]` directly as types, I don't need to rely on two sets of conditionals (the ternary above and this additional check) to get the parser type right.

Additionally, this was an interesting case, because these three commands specifically justified the `multi: true` in the `CommandOption` struct, but we can see here that we weren't using that information do the filtering and instead were hard-coding it.

The updated typing eliminates both issues.